crates/cli/src/bin/wasm-bindgen-test-runner/server.rs

```diff
@@ -218,6 +244,25 @@ pub fn spawn(
                 )
             };
             return set_isolate_origin_headers(Response::from_data("text/html", s));
+        } else if request.url() == "/__coverage/dump" {
+            let profraw_path = coverage.as_ref().expect(
+                "Received coverage dump request but server wasn't set up to accept coverage",
+            );
+            // This is run after all tests are done and dumps the data received in the request
+            // into a single profraw file
+            let mut profraw =
+                std::fs::File::create(profraw_path).expect("Couldn't create .profraw for coverage");
+            let mut data = Vec::new();
+            request
+                .data()
+                .expect("Expected coverage data in body")
+                .read_to_end(&mut data)
+                .expect("Failed to read message body");
+
+            profraw
+                .write_all(&data)
+                .expect("Couldn't dump coverage data to profraw");
+            return Response::text("Coverage dumped");
```

Comment:
We probably want to log an error instead of panicking here.
Might be simpler to move this block into it's own function returning a `Result` and handling logging at the call site.